src/app/parcels/batch/BatchParcelDataGrid.tsx

```diff
@@ -0,0 +1,267 @@
+    {
+        field: "otherItems",
+        headerName: "Other Items",
+        width: 200,
+        editable: true,
+    },
+    {
+        field: "deliveryInstructions",
+        headerName: "Delivery Instructions",
+        width: 200,
+        editable: true,
+    },
+    {
+        field: "extraInformation",
+        headerName: "Extra Information",
+        width: 200,
+        editable: true,
+    },
+
+    {
+        field: "attentionFlag",
+        headerName: "Attention Flag",
+        width: 150,
+        editable: true,
+    },
+    {
+        field: "signpostingCall",
+        headerName: "Signposting Call",
+        width: 150,
+        editable: true,
+    },
+    {
+        field: "notes",
+        headerName: "Notes",
+        width: 200,
+        editable: true,
+    },
+    {
+        field: "voucherNumber",
+        headerName: "Voucher Number",
+        width: 150,
+        editable: true,
+    },
+    {
+        field: "packingDate",
+        headerName: "Packing Date",
+        width: 150,
+        editable: true,
+    },
+    {
+        field: "packingSlot",
+        headerName: "Packing Slot",
+        width: 150,
+        editable: true,
+    },
+    {
+        field: "shippingMethod",
+        headerName: "Shipping Method",
+        width: 150,
+        editable: true,
+    },
+    {
+        field: "collectionInfo",
+        headerName: "Collection Info",
+        width: 200,
+        editable: true,
+    },
+];
+
+const styledColumns: GridColDef<(typeof mockData)[number]>[] = columns.map((column) => {
+    return { ...column, headerAlign: "center", align: "center" };
+});
+
+const mockData: BatchGridDisplayRow[] = Array.from({ length: 6 }, (_, index) => ({
+    id: index,
+    rowName: index,
+    fullName: "",
+    phoneNumber: "",
+    address: "",
+    adults: null,
+    children: null,
+    listType: "",
+    dietaryRequirements: "",
+    feminineProducts: "",
+    babyProducts: "",
+    petFood: "",
+    otherItems: "",
+    deliveryInstructions: "",
+    extraInformation: "",
+    attentionFlag: "",
+    signpostingCall: "",
+    notes: "",
+    voucherNumber: "",
+    packingDate: "",
+    packingSlot: "",
+    shippingMethod: "",
+    collectionInfo: "",
+}));
+
+mockData[1] = {
+    ...mockData[1],
+    fullName: "John Doe",
+    phoneNumber: "0123456789",
+    address: "123 Main Street, Anytown, USA",
+    adults: 3,
+    children: 1,
+    listType: "Hotel",
+    dietaryRequirements: "Vegetarian",
+    feminineProducts: "Yes, tampons",
+    babyProducts: "Yes, nappySize = 7",
+    petFood: "No",
+    otherItems: "No",
+    deliveryInstructions: "N/A",
+    extraInformation: "No",
+    attentionFlag: "No",
+    signpostingCall: "No",
+    notes: "No",
+    voucherNumber: "1234567890",
+    packingDate: "2022-01-01",
+    packingSlot: "AM",
+    shippingMethod: "Collection",
+    collectionInfo: "Test centre, date, slot",
+};
+
+interface BatchParcelDataGridProps {
+    rows?: BatchGridDisplayRow[];
+}
```

Comment:
I might have missed something, but we probably want rows to always be defined and to just be an empty array if there's nothing to show.